File: mk/trust-policy.ts
#!/usr/bin/env -S node --loader tsm
import { writeFileSync } from "node:fs";

import { printESM, versec } from "@ndn/trust-schema";

const policy = versec.load(`
_network: "ndn" | "yoursunny"
_sitename: s1 | (s1/s2) | (s1/s2/s3)
_routername: _network/_sitename/"%C1.Router"/routerid

rootcert: _network/_CERT
sitecert: _network/_sitename/_CERT
operatorcert: _network/_sitename/"%C1.Operator"/opid/_CERT
routercert: _routername/_CERT
lsdbdata: _routername/"nlsr"/"lsdb"/lsatype/version/segment

lsdbdata <= routercert <= operatorcert <= sitecert <= rootcert
_CERT: "KEY"/_/_/_
`);

writeFileSync("src/model/trust-policy.ts", printESM(policy));

File: mk/trust-policy.rs
// compile to .tlv
// 1. activate python-ndn venv
// 2. python ~/NDNts/pkg/lvs/compile.py <mk/trust-policy.rs >mk/trust-policy.tlv

#network: network & { network: "ndn" | "yoursunny" }
#sitename: s10
#sitename: s20/s21
#sitename: s30/s31/s32
#routername: #network/#sitename/"%C1.Router"/routerid

#rootcert: #network/#CERT
#sitecert: #network/#sitename/#CERT <= #rootcert
#operatorcert: #network/#sitename/"%C1.Operator"/opid/#CERT <= #sitecert
#routercert: #routername/#CERT <= #operatorcert
#lsdbdata: #routername/"nlsr"/"lsdb"/lsatype/version/segment & { lsatype: "names" | "adjacencies" | "coordinates" } <= #routercert

#CERT: "KEY"/_/_/_
